examples/equilibria/anaylsis_toolbox_examples.ex.py

```diff
@@ -0,0 +1,251 @@
+#
+# **Note:**
+#
+# Both EqAnalysis and MultiEqAnalysis can take equilibria of different types
+# (i.e, fixed or free), and/or with different grid sizes and grid resolutions.
+# However, it should be noted that not all of the avaible tools can be applied
+# to the fixed boundary equilibria.
+
+# %% [markdown]
+# #### Inputs
+#
+# We are going to use MAST-U-like and DEMO-like equilibria
+# as the inputs for our examples.
+
+# %%
+# MAST-U-like
+masty_path = Path("../../tests/equilibria/test_data/SH_test_file.json")
+masty_eq = select_eq(masty_path)
+# DEMO-like-SN
+single_demoish_path = Path("../../tests/equilibria/test_data/eqref_OOB.json")
+single_demoish_eq = select_eq(single_demoish_path, from_cocos=7)
+# DEMO-like-DN
+double_demoish_path = Path("../../tests/equilibria/test_data/DN-DEMO_eqref.json")
+double_demoish_eq = select_eq(double_demoish_path)
+
+# %% [markdown]
+# #### EqAnalysis
+#
+# Fisrt we will look at the untilities avilable in the EqAnalysis class
+# for comparision to a reference equilibria.
+#
+# The EqDiagnosticOptions class contains the information needed to
+# set up the comparison plots used during an optimisation or
+# (as is the case here) for anaylsis purposes.
+#
+# We will use the single null DEMO-like equilibrium as our reference equilibrium.
+
+# %%
+# Set up to look at the psi difference
+diag_ops_1 = EqDiagnosticOptions(
+    psi_diff=PsiPlotType.PSI_DIFF,
+    split_psi_plots=EqSubplots.XZ,
+    reference_eq=single_demoish_eq,
+)
+# Set up to look at the absolute psi difference seperately
+# for plamsa and coilset contributions
+diag_ops_2 = EqDiagnosticOptions(
+    psi_diff=PsiPlotType.PSI_ABS_DIFF,
+    split_psi_plots=EqSubplots.XZ_COMPONENT_PSI,
+    reference_eq=single_demoish_eq,
+    lcfs_mask=LCFSMask.IN,
+)
+
+# %%
+# Here we create our anaylsis class with the MAST-U-like and
+# the double null DEMO-like equilibria.
+eq_analysis_1 = EqAnalysis(diag_ops_1, double_demoish_eq)
+eq_analysis_1b = EqAnalysis(diag_ops_1, masty_eq)
+eq_analysis_2 = EqAnalysis(diag_ops_2, double_demoish_eq)
+
+# %%
+# Plot equilibrium
+eq_analysis_1.plot()
+
+# %%
+# Plot field components
+ax = eq_analysis_1.plot_field()
+
+# %%
+# Plot equilibrium normalised profiles
+eq_analysis_1.plot_profiles()
+
+# %%
+# Both together
+ax = eq_analysis_1.plot_equilibria_with_profiles()
+
+# %%
+# Plot 1-D profiles at magnetic axis
+eq_analysis_1.plot_eq_core_mag_axis()
+
+# %%
+# PLot comparison of equilibrium profiles to reference equilibrium profiles
+eq_analysis_1.plot_compare_profiles()
+
+# %%
+eq_analysis_1b.plot_compare_profiles()
+
+# %%
+# Plot an assortment of physics parameters for the plamsa core
+# Note that the dataclass with the results is also output
+core_results, ax = eq_analysis_1.plot_eq_core_analysis(ax=None)
+
+# %%
+# Key parameters as a table
+eq_analysis_1.physics_info_table()
+
+# %%
+# PLot the equilibrium and reference equilibrium seperatrices
+ax = eq_analysis_1.plot_compare_separatrix()
```

Comment:
Could we get axis equal set on this figure, it seems at least locally thats not been set in some instances although I can see it in xz_plot_setup.